pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentTripleLong2LongHashMap.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.util.collections;
+
+import java.util.HashMap;
+
+public class ConcurrentTripleLong2LongHashMap {
+    public class TripleLong{
+        public long first;
+        public long second;
+        public long third;
```

Comment:
Is this intentionally mutable?

Reply:
`TripleLong` will be removed after i implement the ConcurrentTripleLong2LongHashMap, as it use wrapper too.
Ideally, we will store the key and value in the array without any wrapper.

Reply:
If FastUtil provide a TripleLong2LongMap, then it has the chance to be better than our implementation.
As the documentation of FastUtil point out, we should test the util class it provide before use it, there are high chance that it is not a good solution.

Reply:
The triple key is most likely not needed and could be solved with a hierarchical datastructure (map of maps). Please see the other comment https://github.com/apache/pulsar/pull/23600#issuecomment-2475922541.

Reply:
There could always be a higher level abstraction to hide the implementation details.